tasks/update-granules-metadata/tests/test-index.js

```diff
@@ -0,0 +1,135 @@
+    s3().createBucket({ Bucket: t.context.protectedBucket }).promise(),
+    s3().createBucket({ Bucket: t.context.systemBucket }).promise(),
+  ]);
+  process.env.system_bucket = t.context.systemBucket;
+  process.env.stackName = t.context.stackName;
+  putJsonS3Object(
+    t.context.systemBucket,
+    getDistributionBucketMapKey(t.context.stackName),
+    {
+      [t.context.stagingBucket]: t.context.stagingBucket,
+      [t.context.publicBucket]: t.context.publicBucket,
+      [t.context.protectedBucket]: t.context.protectedBucket,
+      [t.context.systemBucket]: t.context.systemBucket,
+    }
+  );
+
+  const payloadPath = path.join(__dirname, 'data', 'payload.json');
+  const rawPayload = fs.readFileSync(payloadPath, 'utf8');
+  t.context.payload = JSON.parse(rawPayload);
+  const filesToUpload = granulesToFileURIs(t.context.payload.input.granules);
+  t.context.filesToUpload = filesToUpload.map((file) =>
+    buildS3Uri(`${t.context.stagingBucket}`, parseS3Uri(file).Key));
+  process.env.REINGEST_GRANULE = false;
+});
+
+test.afterEach.always(async (t) => {
+  await recursivelyDeleteS3Bucket(t.context.publicBucket);
+  await recursivelyDeleteS3Bucket(t.context.stagingBucket);
+  await recursivelyDeleteS3Bucket(t.context.protectedBucket);
+  await recursivelyDeleteS3Bucket(t.context.systemBucket);
+});
+
+test.serial('Should add etag to each CMR metadata file by checking that etag is one of more characters not whitespace', async (t) => {
+  const newPayload = buildPayload(t);
+  const filesToUpload = cloneDeep(t.context.filesToUpload);
+  await uploadFiles(filesToUpload, t.context.stagingBucket);
+
+  const output = await updateGranulesMetadata(newPayload);
+
+  output.granules.forEach((g) => g.files
+    .filter(isCMRFile)
+    .forEach(({ etag = '' }) => t.regex(etag, /"\S+"/)));
+});
+
+test.serial('Should update existing etag on CMR metadata file', async (t) => {
+  const newPayload = buildPayload(t);
+  const filesToUpload = cloneDeep(t.context.filesToUpload);
+  await uploadFiles(filesToUpload, t.context.stagingBucket);
+
+  const output = await updateGranulesMetadata(newPayload);
+  const granule = output.granules[output.granules.findIndex((g) => g.granuleId === 'MOD11A1.A2017200.h19v04.006.2017201090722')];
+  const fileWithPrevEtag = granule.files.filter(isCMRFile);
+  t.true(fileWithPrevEtag !== '13f2bb38e22496fe9d42e761c42a0e67');
+});
+
+test.serial('Updated-granules-metadata throws an error when cmr file type is distribution and no distribution endpoint is set', async (t) => {
+  const newPayload = buildPayload(t);
+  delete newPayload.config.distribution_endpoint;
+
+  const filesToUpload = cloneDeep(t.context.filesToUpload);
+  await uploadFiles(filesToUpload, t.context.stagingBucket);
+
+  const error = await t.throwsAsync(
+    () => updateGranulesMetadata(newPayload)
+  );
+
+  t.is(error.message, 'cmrGranuleUrlType is distribution, but no distribution endpoint is configured.');
```

Comment:
I know that this is how the test is already written, but it can be refactored to now that we're on a later version of `ava`:

```
  const error = await t.throwsAsync(
    () => updateGranulesMetadata(newPayload),
    { message: 'cmrGranuleUrlType is distribution, but no distribution endpoint is configured.' } 
  );
```